lib/model/rofolder.go

```diff
@@ -16,113 +16,135 @@ import (
 
 type roFolder struct {
 	stateTracker
+	scan
 
-	folder    string
-	intv      time.Duration
-	timer     *time.Timer
-	model     *Model
-	stop      chan struct{}
-	scanNow   chan rescanRequest
-	delayScan chan time.Duration
+	folderId string
+	model    *Model
+	stop     chan struct{}
 }
 
 type rescanRequest struct {
-	subs []string
-	err  chan error
+	subdirs []string
+	err     chan error
 }
 
-func newROFolder(model *Model, folder string, interval time.Duration) *roFolder {
+// bundle all folder scan activity
+type scan struct {
+	scanInterval time.Duration
+	scanTimer    *time.Timer
+	scanNow      chan rescanRequest
+	scanDelay    chan time.Duration
+}
+
+func (scan *scan) rescheduleScan() {
```

Comment:
In this declaration, and in `func (folder *folder) ...` and similar, it's not customary to have a long variable name for the method receiver. A single character abbreviation is idiomatic and appropriate as it's the variable that the method necessarily revolves around and the first one that enters the reader's consciousness. Hence `func (f *folder) ...` and `func (s *scan) ...` and so on.